Project2015To2017/Reading/ProjectPropertiesReader.cs

```diff
@@ -1,180 +1,158 @@
+			project.BuildEvents = new[] {project.PrimaryPropertyGroup}.Concat(project.AdditionalPropertyGroups)
+				.Elements()
+				.Where(x =>
+					x.Name.LocalName == "PostBuildEvent" ||
+					x.Name.LocalName == "PreBuildEvent")
 				.ToArray();
-			project.AdditionalPropertyGroups = ReadAdditionalPropertyGroups(project, propertyGroups);
 
-			project.Imports = projectXml.Element(project.XmlNamespace + "Project")
-				.Elements(project.XmlNamespace + "Import").Where(x =>
-					x.Attribute("Project")?.Value != @"$(MSBuildToolsPath)\Microsoft.CSharp.targets" &&
-					x.Attribute("Project")?.Value != @"$(MSBuildBinPath)\Microsoft.CSharp.targets" &&
-					x.Attribute("Project")?.Value !=
-					@"$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props").ToArray();
-
-			project.Targets = projectXml.Element(project.XmlNamespace + "Project")
-				.Elements(project.XmlNamespace + "Target").ToArray();
+			project.Imports = project.ProjectDocument.Root
+				.Elements(project.XmlNamespace + "Import")
+				.Where(x => !RemoveMSBuildImports.Contains(x.Attribute("Project")?.Value))
+				.ToArray();
 
-			if (project.Type == ApplicationType.Unknown)
-			{
-				throw new NotSupportedException("Unable to parse output type.");
-			}
+			project.Targets = project.ProjectDocument.Root
+				.Elements(project.XmlNamespace + "Target")
+				.ToArray();
 		}
 
-		private static (List<string>, List<string>) ReadConditionals(XElement[] unconditionalPropertyGroups,
-			Project project)
+		private static (List<string>, List<string>) ReadConfigurationPlatformVariants(Project project)
 		{
-			var projectXml = project.ProjectDocument;
-
 			var configurationSet = new HashSet<string>();
 			var platformSet = new HashSet<string>();
 
 			var configurationsFromProperty = ParseFromProperty("Configurations");
 			var platformsFromProperty = ParseFromProperty("Platforms");
 
-			var needConfigurations = configurationsFromProperty == null;
-			if (!needConfigurations)
+			if (configurationsFromProperty != null)
 			{
 				foreach (var configuration in configurationsFromProperty)
 				{
 					configurationSet.Add(configuration);
 				}
 			}
-
-			var needPlatforms = platformsFromProperty == null;
-			if (!needPlatforms)
+			else
 			{
-				foreach (var platform in platformsFromProperty)
-				{
-					platformSet.Add(platform);
-				}
+				configurationSet.Add("Debug");
+				configurationSet.Add("Release");
 			}
 
-			if (project.IsModernProject)
+			if (platformsFromProperty != null)
 			{
-				if (needConfigurations)
-				{
-					configurationSet.Add("Debug");
-					configurationSet.Add("Release");
-					needConfigurations = false;
-				}
-
-				if (needPlatforms)
+				foreach (var platform in platformsFromProperty)
 				{
-					platformSet.Add("AnyCPU");
-					needPlatforms = false;
+					platformSet.Add(platform);
 				}
 			}
-
-			if (needConfigurations || needPlatforms)
+			else
 			{
-				foreach (var x in projectXml.Descendants())
```

Comment:
I decided that there is no need to traverse the XML tree. Neither did it match MSBuild behavior.

Reply:
We emit diagnostic warning in mismatch cases.

Reply:
Lot of red here, I like it 👍